examples/ex2-dot-product.cpp

```diff
@@ -92,33 +92,43 @@ int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
 
   std::cout << "\n Running RAJA sequential dot product...\n";
 
-  RAJA::ReduceSum<RAJA::seq_reduce, double> seqdot(0.0);
+  RAJA::ReduceSum<RAJA::seq_reduce, double> seqdot;
 
   RAJA::forall<RAJA::seq_exec>(RAJA::RangeSegment(0, N), [=] (int i) { 
-    seqdot += a[i] * b[i]; 
+      seqdot += a[i] * b[i]; 
   });
-
+  
   dot = seqdot.get();
   std::cout << "\t (a, b) = " << dot << std::endl;
 
   checkResult(dot, dot_ref);
 
+  std::cout<<"reset the data "<<std::endl;
+  seqdot.reset();
+  dot = seqdot.get();
+  std::cout << "\t (a, b) = " << dot << std::endl;
+
+
+
 
 //----------------------------------------------------------------------------//
 
 #if defined(RAJA_ENABLE_OPENMP)
   std::cout << "\n Running RAJA OpenMP dot product...\n";
 
-  RAJA::ReduceSum<RAJA::omp_reduce, double> ompdot(0.0);
+  RAJA::ReduceSum<RAJA::omp_reduce, double> ompdot;
 
   RAJA::forall<RAJA::omp_parallel_for_exec>(RAJA::RangeSegment(0, N), [=] (int i) { 
     ompdot += a[i] * b[i]; 
   });    
 
   dot = ompdot.get();
   std::cout << "\t (a, b) = " << dot << std::endl;
-
   checkResult(dot, dot_ref);
+
```

Comment:
Why is this in the example?  We should be checking this sort of thing in the unit tests.

Reply:
@rhornung67, it is still a work in progress. I am working on the CUDA version at the moment. I modified the example to tryout out my modifications, I'll restore the original and put together the unit-tests.

Reply:
OK. No problem.